crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -130,10 +130,19 @@ impl Solution {
 
         // The solution is to do access list estimation in two steps: first, simulate
         // moving 1 wei into every smart contract to get a partial access list.
-        let partial_access_lists = try_join_all(
-            self.user_trades()
-                .map(|trade| async { self.partial_access_list(eth, simulator, trade).await }),
-        )
+        let partial_access_lists = try_join_all(self.user_trades().map(|trade| async {
+            if !trade.order.buys_eth() || !trade.order.pays_to_contract(eth).await? {
+                return Ok(Default::default());
+            }
```

Comment:
Usually you can't make meaningful progress after an error. But in this case when this call fails for an order that doesn't pay out to a smart contract we could take our chances and assume it doesn't pay out to a smart contract and use a default access list there.
If we are right we save the `Settlement` from erroring out and if we are wrong the settlement will fail again when we try to estimate the gas (because we didn't generate the required access list after the error) so no harm done IMO.